crates/cxx-qt-gen/src/generator/cpp/mod.rs

```diff
@@ -78,6 +78,38 @@ impl GeneratedCppBlocks {
         })
     }
 }
+/// Returns a vector of the names and types ([CppNamedType] of the parameters of this method, used in cpp generation step
+pub fn get_cpp_params(method: &ForeignItemFn, type_names: &TypeNames) -> Result<Vec<CppNamedType>> {
+    method
+        .sig
+        .inputs
+        .iter()
+        .map(|input| {
+            // Match parameters to extract their idents
+            if let FnArg::Typed(PatType { pat, ty, .. }) = input {
+                let ident = if let Pat::Ident(PatIdent { ident, .. }) = &**pat {
+                    ident
+                } else {
+                    return Err(Error::new(input.span(), "Unknown pattern for type"));
+                };
+
+                // If the name of the argument is self then ignore,
+                // as this is likely the self: Pin<T>
+                if ident == "self" {
+                    Ok(None)
+                } else {
+                    Ok(Some(CppNamedType {
+                        ident: ident.to_string(),
+                        ty: syn_type_to_cpp_type(ty, type_names)?,
+                    }))
+                }
+            } else {
+                Ok(None)
+            }
+        })
+        .filter_map(|result| result.map_or_else(|e| Some(Err(e)), |v| v.map(Ok)))
+        .collect()
+}
 
 mod utils;
```

Comment:
this mod should probably be near the top with the others ?

Reply:
Done